decoder/api_pokemon_common.go

```diff
@@ -0,0 +1,61 @@
+package decoder
+
+import (
+    "github.com/UnownHash/gohbem"
+)
+
+func buildApiPokemonResult(pokemon Pokemon) ApiPokemonResult {
```

Comment:
Just take a pointer here (ie `pokemon *Pokemon`) and it looks perfect. It could return a pointer also I guess since it's using it as a pointer (and returning the pointer gets deserialised also)

Reply:
Should be good now 🙂